src/lib.rs

```diff
@@ -311,263 +333,246 @@ impl<'a, Writer: Write> Printer<'a, Writer> {
-                        25,
-                        self.border_style.inner_sep()
-                    )?;
-                }
-                write!(
-                    &mut self.buffer_line,
-                    "{2}{0:1$}",
-                    "",
-                    8,
-                    self.border_style.outer_sep()
-                )?;
-                for _ in 0..self.panels - 1 {
-                    write!(
-                        &mut self.buffer_line,
-                        "{2}{0:1$}",
-                        "",
-                        8,
-                        self.border_style.inner_sep()
-                    )?;
-                }
-                writeln!(&mut self.buffer_line, "{}", self.border_style.outer_sep())?;
-                self.writer.write_all(&self.buffer_line)?;
-            }
-            return Ok(());
+    pub fn print_char_panel(&mut self) -> io::Result<()> {
+        for i in 0..self.line_buf.len() {
+            self.print_char(i as u64)?;
         }
+        Ok(())
+    }
 
-        let squeeze_action = self.squeezer.action();
-
-        // print empty space on last line
-        if squeeze_action != SqueezeAction::Delete {
-            if len < usize::from(8 * self.panels) {
-                write!(&mut self.buffer_line, "{0:1$}", "", 3 * (8 - len % 8))?;
-                for _ in 0..(usize::from(8 * self.panels) - (len + (8 - len % 8))) / 8 {
-                    write!(
-                        &mut self.buffer_line,
-                        "{2}{0:1$}",
-                        "",
-                        1 + 3 * 8,
-                        self.border_style.inner_sep()
-                    )?;
-                }
-            }
-            write!(&mut self.buffer_line, "{}", self.border_style.outer_sep())?;
+    fn print_byte(&mut self, i: usize, b: u8) -> io::Result<()> {
+        match self.squeezer {
+            Squeezer::Print | Squeezer::Delete => self.writer.write_all(b"   ")?,
+            Squeezer::Ignore | Squeezer::Disabled => self
+                .writer
+                .write_all(self.byte_hex_panel[b as usize].as_bytes())?,
         }
-        self.print_char_panel();
-
-        match squeeze_action {
-            SqueezeAction::Print => {
-                self.buffer_line.clear();
-                let style = COLOR_OFFSET.normal();
-                let asterisk = if self.show_color {
-                    format!("{}", style.paint("*"))
-                } else {
-                    String::from("*")
-                };
-
-                write!(
-                    &mut self.buffer_line,
-                    "{3}{0}{1:2$}{3}",
-                    asterisk,
-                    "",
-                    7,
-                    self.border_style.outer_sep()
+        // byte is last in panel
+        if i % 8 == 7 {
+            // byte is last in last panel
+            if i as u64 % (8 * self.panels) == 8 * self.panels - 1 {
+                self.writer.write_all(b" ")?;
```

Comment:
Maybe we can resolve the clippy suggestions here (all if blocks contain the same code at the start)